Signal/src/phone/PhoneManager.m

```diff
@@ -139,7 +141,16 @@ - (void)incomingCallWithSession:(ResponderSessionDescriptor *)session {
 
 - (BOOL)hasOngoingCall
 {
-    return self.curCallController != nil;
+    if (!self.curCallController) {
+        return NO;
+    }
+
+    return [self.curCallController callState].futureTermination.isIncomplete;
```

Comment:
https://github.com/WhisperSystems/Signal-iOS/pull/1698 has been closed. I've tested it and it works properly in every scenario I tried.